opentelemetry-sdk/tests/metrics/test_aggregation.py

```diff
@@ -308,3 +311,280 @@ def test_collect(self):
+            ),
+            current_point,
+        )
+
+    def test_current_point_sum_aggregation_temporality_delta(self):
+
+        self.assertEqual(
+            _convert_aggregation_temporality(
+                Sum(
+                    start_time_unix_nano=1,
+                    time_unix_nano=2,
+                    value=3,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=False,
+                ),
+                Sum(
+                    start_time_unix_nano=4,
+                    time_unix_nano=5,
+                    value=6,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=False,
+                ),
+                AggregationTemporality.DELTA,
+            ),
+            Sum(
+                start_time_unix_nano=2,
+                time_unix_nano=5,
+                value=3,
+                aggregation_temporality=AggregationTemporality.DELTA,
+                is_monotonic=False,
+            ),
+        )
+
+        self.assertEqual(
+            _convert_aggregation_temporality(
+                Sum(
+                    start_time_unix_nano=1,
+                    time_unix_nano=2,
+                    value=3,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=True,
+                ),
+                Sum(
+                    start_time_unix_nano=4,
+                    time_unix_nano=5,
+                    value=6,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=False,
+                ),
+                AggregationTemporality.DELTA,
+            ),
+            Sum(
+                start_time_unix_nano=2,
+                time_unix_nano=5,
+                value=3,
+                aggregation_temporality=AggregationTemporality.DELTA,
+                is_monotonic=False,
+            ),
+        )
+
+        self.assertEqual(
+            _convert_aggregation_temporality(
+                Sum(
+                    start_time_unix_nano=1,
+                    time_unix_nano=2,
+                    value=3,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=True,
+                ),
+                Sum(
+                    start_time_unix_nano=4,
+                    time_unix_nano=5,
+                    value=6,
+                    aggregation_temporality=AggregationTemporality.CUMULATIVE,
+                    is_monotonic=True,
+                ),
+                AggregationTemporality.DELTA,
+            ),
+            Sum(
+                start_time_unix_nano=2,
+                time_unix_nano=5,
+                value=3,
+                aggregation_temporality=AggregationTemporality.DELTA,
+                is_monotonic=True,
+            ),
```

Comment:
Not able to understand this data, can you please explain? Why does the second point start at 4?

Reply:
This intends to test the algorithm, for these purposes a different number is used in every time field to make it easier to compare the actual results to the expected ones. This also makes it obvious which values are used in the algorithm and which ones are not.

Reply:
@lonewolf3739 can you please review @ocelotl's comment.